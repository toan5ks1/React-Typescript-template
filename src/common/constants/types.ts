import { FieldErrors } from "react-hook-form";
import {
    CoursesManyQuery,
    LocationListByIdsQuery,
    StudentsManyQuery,
    TeacherManyQuery,
} from "src/services/bob/bob-types";
import { Lesson_LessonReportListByLessonIdsQuery } from "src/squads/lesson/service/bob/bob-types";

import { Paging } from "manabuf/common/v1/requests_pb";
import { OpeningStatus } from "manabuf/eureka/v1/scheduler_pb";

export interface DateCommon {
    date: string | Date | number;
}

export interface OptionSelectType {
    id: string | number;
    value: string;
    label?: string | JSX.Element; // add a label for more descriptive item when map to element
    disabled?: boolean;
}

export type ErrorMessageArrayFieldHF<T> =
    | { message: string }
    | (FieldErrors<T> | undefined)[]
    | undefined;

export type ArrayElement<ArrayType extends readonly unknown[]> =
    ArrayType extends readonly (infer ElementType)[] ? ElementType : never;

export interface AutocompleteRecord {
    [key: string]: any;
}

//Form filter advanced types
export type FilterAppliedObjects<T> = {
    name: string;
    chipLabel?: string;
    inputLabel: string;
    isApplied: boolean;
    defaultValue: T;
};

export type FilterAppliedObjectsMap<T> = {
    [X in keyof T]: FilterAppliedObjects<T[X]>;
};

export type FilterAppliedObjectValuesMap<T> = FilterAppliedObjects<T[keyof T]>[];

export interface PagingRequest extends Paging.AsObject {}

export type StudentManyType = ArrayElement<StudentsManyQuery["users"]>;

export type CourseManyType = ArrayElement<CoursesManyQuery["courses"]>;

export type TeacherManyType = ArrayElement<TeacherManyQuery["find_teacher_by_school_id"]>;

export type LessonReportManyType = ArrayElement<
    Lesson_LessonReportListByLessonIdsQuery["lesson_reports"]
>;

export type FilterDateType = null | string | Date;

export type CourseObjectType = Record<CourseManyType["course_id"], CourseManyType["name"]>;

export type LessonReportObjectType = Record<
    LessonReportManyType["lesson_report_id"],
    LessonReportManyType["report_submitting_status"]
>;

export type TeacherObjectType = Record<TeacherManyType["user_id"], TeacherManyType["name"]>;

export type StudentObjectType = Record<StudentManyType["user_id"], StudentManyType["name"]>;

export type StudentSelectType = Pick<StudentManyType, "user_id" | "name">;

export type TeacherSelectType = Pick<TeacherManyType, "user_id" | "name">;

export type CourseSelectType = Pick<CourseManyType, "course_id" | "name">;

export type LocationManyType = ArrayElement<LocationListByIdsQuery["locations"]>;

export type LocationObjectType = Record<LocationManyType["location_id"], LocationManyType["name"]>;

export interface FilterParams<T> {
    keyword: string;
    filter?: T;
}

export interface QueryOpeningTimeProps {
    start_time: string;
    end_time: string;
    brand_id?: string;
    center_id?: string;
    scheduler_type: string;
}

export type CreateOpeningTimeFormProps = {
    brand_id?: string;
    center_id?: string;
    opening_status: OpeningStatus;
    opening_date: Date | undefined;
    opening_hour: number;
    opening_minutes: number;
    repeat_option: number;
    date_until?: Date;
};

export interface CalendarEventProps {
    id?: string;
    allDay?: boolean;
    start: Date;
    end: Date;
    metaData: {
        title?: string;
        description?: string;
        remark?: string;
    };
    brandId: string;
    centerId: string;
    schedulePattern: {
        repeatOptionData: {
            repeatType: string; // todo: change to RepeatType enum
            count?: Number;
            until?: Date;
        };
    };
}
