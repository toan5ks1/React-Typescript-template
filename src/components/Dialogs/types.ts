import { ReactElement, ReactNode } from "react";

import { UseFormMethods } from "src/typings/react-hook-form";

import { Breakpoint } from "@mui/material";
import { ButtonBaseProps } from "src/components/Buttons/ButtonBase";

import { WrapperPageContentProps } from "../Wrappers/WrapperPageContent";
import { DialogBaseProps } from "./DialogBase";

export type DialogFooterType = ReactElement | "portal" | "empty" | "default";
export interface DialogWithHeaderFooterProps extends DialogBaseProps {
    onEdit?: () => void;
    onDelete?: () => void;
    onClose: () => void;
    onSave?: () => void;
    onCloseClick?: () => void;
    maxWidthBox?: Breakpoint;
    minWidthBox?: Breakpoint;
    header?: ReactElement;
    footer?: DialogFooterType;
    textClose?: string;
    textSave?: string;
    footerConfirmButtonProps?: {
        variant?: ButtonBaseProps["variant"];
        color?: ButtonBaseProps["color"];
        disabled?: ButtonBaseProps["disabled"];
    };
    shouldDisableClosing?: boolean;
    shouldShowCancelButton?: boolean;
    additionalHeights?: number;
    shouldPressKey?: boolean;
}

export interface DialogFullScreenProps extends DialogBaseProps {
    onSave?: () => void;
    onClose: () => void;
    textClose?: string;
    textSave?: string;
    footerConfirmButtonProps?: {
        variant?: ButtonBaseProps["variant"];
        color?: ButtonBaseProps["color"];
        disabled?: ButtonBaseProps["disabled"];
    };
    footer?: ReactElement;
    title: string;
    isShowingBackdrop?: boolean;
    contentSize?: WrapperPageContentProps["variant"];
    dialogCancelConfirmProps?:
        | Pick<DialogCancelConfirmProps, "title" | "textCancelDialog" | "textSave">
        | boolean;
}

export type DialogFooterProps = Pick<
    DialogWithHeaderFooterProps | DialogFullScreenProps,
    "onClose" | "onSave" | "textClose" | "textSave" | "footerConfirmButtonProps"
>;

export interface DialogHeaderProps
    extends Pick<
        DialogWithHeaderFooterProps,
        "onClose" | "onDelete" | "onEdit" | "title" | "onCloseClick" | "shouldDisableClosing"
    > {
    children?: ReactNode;
    disableBottomPadding?: boolean;
}

export interface DialogCancelConfirmProps
    extends Pick<
        DialogWithHeaderFooterProps,
        | "onSave"
        | "title"
        | "onClose"
        | "open"
        | "textClose"
        | "textSave"
        | "footerConfirmButtonProps"
    > {
    textCancelDialog?: ReactNode;
}

export type DialogFooterConfirmProps = Pick<
    DialogWithHeaderFooterProps,
    | "onClose"
    | "onSave"
    | "textClose"
    | "textSave"
    | "footerConfirmButtonProps"
    | "shouldShowCancelButton"
>;

export type DialogFullScreenHeaderProps = Pick<DialogFullScreenProps, "onClose" | "title">;

export interface FooterButtonProps {
    variant?: ButtonBaseProps["variant"];
    color?: ButtonBaseProps["color"];
}

export interface DialogHFProps<T> extends DialogWithHeaderFooterProps, UseFormMethods<T> {}

export interface DialogFullScreenHFProps<T> extends DialogFullScreenProps, UseFormMethods<T> {}
