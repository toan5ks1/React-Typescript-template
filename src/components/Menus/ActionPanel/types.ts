import { ReactElement, ReactNode } from "react";

import { MutationMenus } from "src/common/constants/enum";
import { TypeEntity } from "src/typings/react-admin";

import { MenuProps } from "@mui/material";
import { IconButtonWithAnchorMenuProps } from "src/components/IconButton/IconButtonWithAnchorMenu";

export interface CustomAction<T = MutationMenus> {
    action: T;
    Icon?: ReactElement;
    label?: string;
    withConfirm?: boolean;
    className?: string;
    disabled?: boolean;
    labelOptions?: object;
    onClick?: () => void;
    render?: (label: string, onAction: () => void) => ReactNode;
}

export type Action<T = MutationMenus> = MutationMenus | CustomAction<T>;

export interface GetActionPropsReturn<T = MutationMenus>
    extends Omit<CustomAction<T>, "action" | "Icon"> {
    actionType: T;
}

export interface ActionPanelProps<T extends any = any, TAction = MutationMenus>
    extends Omit<MenuProps, "onClick" | "open"> {
    size?: "inherit" | "default" | "small" | "large";
    record?: T;
    loading?: boolean;
    disables?: {
        [x: string]: boolean;
    };
    disabled?: boolean;
    onAction?: (...args: any[]) => void;
    recordName: ReactNode;
    actions?: Action<TAction>[];
    resource?: TypeEntity | string;
    buttonStyle?: IconButtonWithAnchorMenuProps["variant"];
    suffixDeleteTitle?: string;
    getActionProps?: (action: Action<TAction>) => GetActionPropsReturn<TAction>;
}

export interface WrapperActionPanelProps<T extends any = any, TAction = MutationMenus>
    extends Omit<ActionPanelProps<T, TAction>, "onClick"> {
    open: boolean;
    anchorEl: HTMLElement | null;
    onClose: () => void;
    onAction: (e: string) => void;
    menuProps?: MenuProps;
    onClick?: () => void;
}
