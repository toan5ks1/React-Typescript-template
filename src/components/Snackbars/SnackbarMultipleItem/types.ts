export type CloseReasonTypes = "timeout" | "maxsnack" | "instructed";
export type SnackbarKey = string;
export type SnackbarMessage = string;

export type TransitionCloseHandler = (
    event: React.SyntheticEvent<any> | null,
    reason: CloseReasonTypes,
    key?: SnackbarKey
) => void;

export type TransitionHandler = (node: HTMLElement, key: SnackbarKey) => void;

export interface TransitionHandlerProps {
    onClose: TransitionCloseHandler;
    onExited: TransitionHandler;
}
