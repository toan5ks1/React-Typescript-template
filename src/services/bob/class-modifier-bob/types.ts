import { MediaType as MediaTypeV1 } from "manabuf/bob/v1/enums_pb";

export declare namespace NsBobModifierClassService {
    export interface ConvertMedia {
        mediaList: Array<{
            media_id?: string;
            name?: string;
            resource: string;
            type: keyof typeof MediaTypeV1;
        }>;
    }
}
