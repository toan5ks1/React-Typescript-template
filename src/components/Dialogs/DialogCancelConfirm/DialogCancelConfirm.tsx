import TypographyTextSecondary from "src/components/Typographys/TypographyTextSecondary";

import DialogWithHeaderFooterHF from "../DialogWithHeaderFooterHF";
import { DialogCancelConfirmProps } from "../types";

import useTranslate from "src/hooks/useTranslate";

const DialogCancelConfirm = (props: DialogCancelConfirmProps) => {
    const t = useTranslate();
    const { textCancelDialog = t("ra.common.editedInformationWillBeLostIfLeave"), ...rest } = props;

    const defaultCancelDialog: Partial<DialogCancelConfirmProps> = {
        title: t("ra.common.cancelDialogTitle"),
        open: false,
        textSave: t("ra.common.action.confirm"),
        textClose: t("ra.common.action.cancel"),
    };

    return (
        <DialogWithHeaderFooterHF
            data-testid="DialogCancelConfirm__dialog"
            footerConfirmButtonProps={{
                color: "error",
                variant: "contained",
            }}
            {...defaultCancelDialog}
            {...rest}
        >
            <TypographyTextSecondary>{textCancelDialog}</TypographyTextSecondary>
        </DialogWithHeaderFooterHF>
    );
};

export default DialogCancelConfirm;
