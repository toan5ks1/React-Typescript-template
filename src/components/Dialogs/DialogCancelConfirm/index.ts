export { default } from "./DialogCancelConfirm";
export * from "./DialogCancelConfirm";
