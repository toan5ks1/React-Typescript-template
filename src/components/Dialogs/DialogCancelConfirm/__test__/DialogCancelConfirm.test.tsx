import { DialogCancelConfirmProps } from "../../types";
import DialogCancelConfirm from "../DialogCancelConfirm";

import { fireEvent, render, RenderResult, screen } from "@testing-library/react";
import TestThemeProvider from "src/test-utils/TestThemeProvider";

describe("<DialogCancelConfirm /> ", () => {
    let wrapper: RenderResult;

    const props: DialogCancelConfirmProps = {
        onClose: jest.fn(),
        onSave: jest.fn(),
        open: true,
        title: "Cancel title",
        textCancelDialog: "Cancel text",
    };

    beforeEach(() => {
        wrapper = render(
            <TestThemeProvider>
                <DialogCancelConfirm {...props} />
            </TestThemeProvider>
        );
    });

    it("should exist title", () => {
        expect(screen.getByText("Cancel title")).toBeInTheDocument();
    });

    it("should exist children", () => {
        expect(screen.getByText("Cancel text")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.getByTestId("DialogCancelConfirm__dialog")).toMatchSnapshot();
    });

    it("should call onSave function", () => {
        const confirmBtn = wrapper.getByTestId("FooterDialogConfirm__buttonSave");

        fireEvent.click(confirmBtn);
        expect(props.onSave).toBeCalled();
    });

    it("should call onClose function", () => {
        const closeBtn = wrapper.getByTestId("FooterDialogConfirm__buttonClose");

        fireEvent.click(closeBtn);
        expect(props.onClose).toBeCalled();
    });
});
