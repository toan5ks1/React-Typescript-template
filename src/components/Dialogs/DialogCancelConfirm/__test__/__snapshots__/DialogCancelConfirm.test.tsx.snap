// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogCancelConfirm />  should match snapshot 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #fff;
  color: #212121;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12);
  margin: 32px;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: calc(100% - 64px);
  max-width: 600px;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

@media (max-width:663.95px) {
  .emotion-3.MuiDialog-paperScrollBody {
    max-width: calc(100% - 64px);
  }
}

.emotion-4 {
  border-radius: 2px;
  box-shadow: 0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12);
  min-width: 600px;
  max-width: 600px;
}

.emotion-5 {
  background-color: #F5F5F5;
}

.emotion-6 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 500;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.6;
  padding: 16px 24px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-7 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 500;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.6;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-12 {
  margin-left: 8px;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 5px;
  font-size: 1.125rem;
  padding: 3px;
}

.emotion-13::-moz-focus-inner {
  border-style: none;
}

.emotion-13.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-13 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-13:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-13:hover {
    background-color: transparent;
  }
}

.emotion-13.Mui-disabled {
  background-color: transparent;
  color: #BDBDBD;
}

.emotion-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-15 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-16 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-overflow-scrolling: touch;
  overflow-y: auto;
  padding: 20px 24px;
  padding: 8px 24px;
  max-height: calc(100vh - 64px - 64px - (52px + 16px + 24px) - 0px);
}

.MuiDialogTitle-root+.emotion-16 {
  padding-top: 0;
}

.emotion-17 {
  padding-top: 8px;
  padding-bottom: 0px;
}

.emotion-18 {
  margin: 0;
  font-size: 1rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.5;
  color: #757575;
}

.emotion-19 {
  padding: 24px;
  padding-top: 16px;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.emotion-21 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #E0E0E0;
  border-radius: 4px;
  color: #212121;
  margin-right: 16px;
}

.emotion-23::-moz-focus-inner {
  border-style: none;
}

.emotion-23.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-23 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-23:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(224, 224, 224, 0.04);
}

@media (hover: none) {
  .emotion-23:hover {
    background-color: transparent;
  }
}

.emotion-23.Mui-disabled {
  color: #BDBDBD;
}

.emotion-23:hover {
  background-color: rgba(33, 33, 33, 0.04);
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #E0E0E0;
  box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),0px 2px 2px rgba(0, 0, 0, 0.14),0px 1px 5px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  color: rgba(0, 0, 0, 0.87);
  background-color: #F44336;
  color: #fff;
}

.emotion-26::-moz-focus-inner {
  border-style: none;
}

.emotion-26.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-26 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-26:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #BDBDBD;
  box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px rgba(0, 0, 0, 0.14),0px 1px 10px rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-26:hover {
    background-color: #E0E0E0;
  }
}

.emotion-26:active {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12);
}

.emotion-26.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px rgba(0, 0, 0, 0.14),0px 1px 18px rgba(0, 0, 0, 0.12);
}

.emotion-26.Mui-disabled {
  color: #BDBDBD;
  box-shadow: none;
  background-color: #E0E0E0;
}

.emotion-26:hover {
  background-color: #E31B0C;
}

<div
  class="MuiModal-root MuiDialog-root emotion-0"
  data-testid="DialogCancelConfirm__dialog"
  role="presentation"
  title="Cancel title"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root emotion-1"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="mui-3"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm emotion-3"
      role="dialog"
    >
      <div
        class="MuiBox-root emotion-4"
        data-testid="DialogWithHeaderFooter_wrapper"
      >
        <div
          class="MuiBox-root emotion-5"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root emotion-6"
            data-testid="DialogWithHeaderFooter__dialogTitle"
            id="mui-3"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-7"
            >
              <div
                class="MuiGrid-root MuiGrid-item emotion-8"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 emotion-9"
                >
                  Cancel title
                </h6>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item emotion-8"
              >
                <div
                  class="MuiBox-root emotion-11"
                >
                  <div
                    class="MuiBox-root emotion-12"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall emotion-13"
                      data-testid="DialogWithHeaderFooter__buttonExit"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                        data-testid="CloseIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root emotion-15"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root emotion-16"
          data-testid="DialogWithHeaderFooter__dialogContent"
        >
          <div
            class="MuiBox-root emotion-17"
          >
            <form
              novalidate=""
            >
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-18"
              >
                Cancel text
              </p>
            </form>
          </div>
        </div>
        <div
          class="MuiBox-root emotion-19"
        >
          <div
            class="MuiDialogActions-root emotion-20"
            data-testid="DialogWithHeaderFooter__dialogActions"
          >
            <div
              class="MuiGrid-root MuiGrid-container emotion-21"
            >
              <div
                class="MuiGrid-root MuiGrid-item emotion-8"
              >
                <button
                  aria-label="ra.common.action.cancel"
                  class="MuiButton-root MuiButton-text MuiButton-textDefault MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root emotion-23"
                  data-testid="FooterDialogConfirm__buttonClose"
                  tabindex="0"
                  type="button"
                >
                  ra.common.action.cancel
                  <span
                    class="MuiTouchRipple-root emotion-15"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item emotion-8"
              >
                <button
                  aria-label="ra.common.action.confirm"
                  class="MuiButton-root MuiButton-contained MuiButton-containedDefault MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-26"
                  data-testid="FooterDialogConfirm__buttonSave"
                  tabindex="0"
                  type="button"
                >
                  ra.common.action.confirm
                  <span
                    class="MuiTouchRipple-root emotion-15"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
